packages/glimmer-runtime/tests/updating-test.ts

```diff
@@ -1793,3 +1793,68 @@ test("HTML namespace is continued to child templates", function() {
   equal(getSvg().namespaceURI, SVG_NAMESPACE);
   equal(getSvg().firstChild.namespaceURI, SVG_NAMESPACE);
 });
+
+QUnit.module("Updating Element Modifiers", {
+  setup: commonSetup
+});
+
+test("Updating a element modifier", assert => {
+  let manager = new TestModifierManager();
+  env.registerInternalModifier('foo', manager);
+
+  let template = compile('<div {{foo bar}}></div>');
+  let input = {
+    bar: 'Super Metroid'
+  };
+
+  render(template, input);
+
+  let valueNode = root.firstChild;
+
+  equalTokens(root, '<div></div>', "initial render");
+  equal(manager.installedElements.length, 1);
+  equal(valueNode, manager.installedElements[0]);
+  equal(manager.updatedElements.length, 0);
+
+  rerender();
+
+  equalTokens(root, '<div></div>', "no change");
+  equal(manager.installedElements.length, 1);
+  equal(valueNode, manager.installedElements[0]);
+  equal(manager.updatedElements.length, 1);
+  equal(valueNode, manager.updatedElements[0]);
+
+  input.bar = 'Super Mario';
+
+  rerender();
+
+  equalTokens(root, '<div></div>', "no change");
+  equal(manager.installedElements.length, 1);
+  equal(valueNode, manager.installedElements[0]);
+  equal(manager.updatedElements.length, 2);
+  equal(valueNode, manager.updatedElements[1]);
+});
+
+test("Const input doesn't trigger update in a element modifier", assert => {
+  let manager = new TestModifierManager();
+  env.registerInternalModifier('foo', manager);
+
+  let template = compile('<div {{foo "bar"}}></div>');
+  let input = {};
+
+  render(template, input);
+
+  let valueNode = root.firstChild;
+
+  equalTokens(root, '<div></div>', "initial render");
+  equal(manager.installedElements.length, 1);
+  equal(valueNode, manager.installedElements[0]);
+  equal(manager.updatedElements.length, 0);
+
+  rerender();
+
+  equalTokens(root, '<div></div>', "no change");
+  equal(manager.installedElements.length, 1);
+  equal(valueNode, manager.installedElements[0]);
+  equal(manager.updatedElements.length, 0);
+});
```

Comment:
@chadhietala I have `"files.trimTrailingWhitespace": true` set in my settings so I think it negates any setting otherwise. It's weird because I'm used to having a newline at the end since that's what my Sublime does. This is annoying.